### PR TITLE
Report ledger information in account_lines RPC (RIPD-1276)

### PR DESCRIPTION
**Pull request by nbougalis**

### CODE REVIEW

**Comment by miguelportilla** [comment]
:+1:

**Comment by willwray** [comment]
👍 
Looks good.

I guess we should also extend the tests in src/ripple/rpc/tests/AccountLinesRPC.test.cpp
Would that fit in this PR?

At first I wasn't sure about the code, as the added loops just copy any rpc error fields. Then I saw that the old 'intentional' assignment to result was previously trashing the lookupLedger result, so all looks good

File: src/ripple/rpc/handlers/AccountLines.cpp [hunk 1]
```diff
@@ -93,9 +93,12 @@ Json::Value doAccountLines (RPC::Context& context)
     std::string strIdent (params[jss::account].asString ());
     AccountID accountID;
 
-    // Intentional assignment in if.  Extra parentheses silence warning.
-    if ((result = RPC::accountFromString (accountID, strIdent)))
+    if (auto jv = RPC::accountFromString (accountID, strIdent))
+    {
+        for (auto it = jv.begin (); it != jv.end (); ++it)
```
Comment (bachase) [inline src/ripple/rpc/handlers/AccountLines.cpp]: Would a range-based for work here?

Comment (vinniefalco) [inline src/ripple/rpc/handlers/AccountLines.cpp]: The problem with range-for is that you don't get the iterator. I'm not sure if the body of that loop needs it though. I think this is solved in the Range TS?

Comment (vinniefalco) [inline src/ripple/rpc/handlers/AccountLines.cpp]: Yeah I think this is an early version of that
```
for(auto it : std::iterator_range(jv))
...
```
http://www.boost.org/doc/libs/1_54_0/libs/range/doc/html/range/reference/utilities/iterator_range.html

Comment (nbougalis) [inline src/ripple/rpc/handlers/AccountLines.cpp]: The iterator is needed inside the body of the loop. I can use `boost::iterator_range` if we want.

Comment (vinniefalco) [inline src/ripple/rpc/handlers/AccountLines.cpp]: That's up to you. Although, I wouldn't bother. I would just wait for the Range TS in the stdlib. Its only a few years away.

Comment (bachase) [inline src/ripple/rpc/handlers/AccountLines.cpp]: Got it. Thanks!

Comment (nbougalis) [inline src/ripple/rpc/handlers/AccountLines.cpp]: I'll pass.

**Comment by nbougalis** [comment]
Merged as 5be33a650dcd4a301280f0c1b7950206e487af2c

**Pull request closed without merging**